### ISSUE TITLE
Stylelint did not report `word-break: break-word` as invalid value

### ISSUE THREAD

**Issue by pkuczynski** [stylelint/stylelint#2907]
Style lint did not report `word-break: break-word` as an invalid value. See https://developer.mozilla.org/en-US/docs/Web/CSS/word-break for valid values.

I can't find a rule which would cover this case but still feels like there should be one recognizing incorrect property values.

To reproduce the issue lint this snippet

```css
.foo {
  word-break: break-word;
}
```

My stylelint version: 8.1.1

I could work on a PR with a new rule, if you think this would be the right approach?

**Comment by ntwb** [comment]
We've started leaning towards using [CSSTree](
https://github.com/csstree/csstree) for this type of validation @pkuczynski 

The [stylelint-csstree-validator](https://github.com/csstree/stylelint-validator#stylelint-csstree-validator) plugin will get you up and running with CSSTree in no time at all 😄 

----

With that said, I tested the above myself and `word-break: break-word;` does *not* return an invalid value error or warning.

It looks like CSSTree add an exception for `break-word` [here](https://github.com/csstree/csstree/blob/c88bae8f8fddca97f13d3037d6de0b8266fe10cc/data/patch.json#L538-L540) that references [this](https://css-tricks.com/almanac/properties/w/word-break/) CSSTricks article

It's there as an exemption for webkit browsers:

> _"Hyphenation is not supported in Chrome. You can break words in Chrome, without breaking Firefox by using the Chrome-specific notation `word-break: break-word`."_

This is also documented on https://caniuse.com/#search=word-break

I've updated the MDN document to add the above info to the end of it as a note 😱

**Comment by pkuczynski** [comment]
Nice catch! Didn't know this one. However, if this is a non-standard value, which can be easily replaced with `word-wrap: break-word`, maybe stylelint should recognize this and report it as a warning?

**Comment by jeddy3** [comment]
> maybe stylelint should recognize this and report it as a warning?

You can use the [flexible "Limit language features" rules](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/rules.md#limit-language-features) to achieve this. In particular, the [`declaration-property-value-blacklist`](https://github.com/stylelint/stylelint/blob/master/lib/rules/declaration-property-value-blacklist/README.md) rule. I believe this series of rules allows a user to disallow nearly all, if not all, languages features of CSS.

**Comment by pkuczynski** [comment]
To be honest I would expect stylelint to _lint_ my styles, which means find all possible bugs and issues. So I think you should either build that functionality into stylelint or include `stylelint-csstree-validator` plugin by default. Don't you think?

**Comment by pkuczynski** [comment]
Which browser @mpyw? According to the docs `break-word` is not a valid value for `word-break`

**Comment by mpyw** [comment]
@pkuczynski Chrome (latest).

**Comment by mpyw** [comment]
I posted a question on the japanese programming community:

[HTML - word-break: break-word; が効いて word-wrap: break-word; overflow-wrap: break-word; が効かない場合とは？｜teratail](https://teratail.com/questions/160226#)
